### PR TITLE
feat: adds cao based cb cluster deployment

### PR DESCRIPTION
**Pull request by boseabhishek**
- initialise operator/admission vers/namespace, crd file using init
- creates couchbase cluster
- lists couchbase clusters
- gets connection info
- uses defaults where applicable

### CODE REVIEW

**Review by Couch-Matt — COMMENTED** [review]
Looks good to me with my limited knowledge of the codebase. Would be nice to have some sort of tutorial video or detailed README so we have a reference for implementing it into FIT or locally.

File: cbdcconfig/cao/2.5/crd.yaml
[the remark this answers is not in the document]
Comment (boseabhishek) [reply cbdcconfig/cao/2.5/crd.yaml:1]: When you do `cbdinocluster init`, it lets you set all operator version, admission version, namespaces, path to crd file and cao bin etc - when you choose `cao` as your preferred deployer

**Comment by boseabhishek** [comment]
Good shout.. can do a doc as part of README

**Comment by brett19** [comment]
@boseabhishek Thanks a lot for this PR.  I used it as a significant basis of integration with k8s, I'm going to leave this open a bit longer until I have an opportunity to merge in the OpenShift parts as well.

**Comment by brett19** [comment]
I've included the OpenShift parts from your code into master now.

**Pull request closed without merging**